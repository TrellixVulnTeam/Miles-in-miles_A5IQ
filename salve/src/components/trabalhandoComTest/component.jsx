import ComponentLink from "../cLink"
import {useState} from 'react'

const LINK = {
  GOOGLE: {
    uri: 'http://google.com.br',
    descricao: 'Google',
  },
  BING: {
    uri: 'http://bing.com.br',
    descricao: 'Bing',
  }
}
export function ComponenteDeTeste(a, b){
  const [link, setLink] = useState(LINK.GOOGLE)

  const numero = 1000
  const booleano = true
  const texto = 'Titulo'

  const google = () => {
    setLink(LINK.GOOGLE)
  }
  const bing = () => {
    setLink(LINK.BING)
  }

  return(
    <div style={{background: 'red', height: '200px', width: '200px', margin: '0 auto', marginTop:'100px', fontSize:'20px', padding:'10px'}}>
      {booleano && (<div>Texto {texto}: <br/> Numero: {numero}</div>)}

      <a href="#!" onMouseEnter={google}>Google | click</a> <br/>
      <a href="#!" onClick={bing}>Bing | click</a> <br/>

      <ComponentLink link={link.uri}>{link.descricao}</ComponentLink>
    </div>
  )
}