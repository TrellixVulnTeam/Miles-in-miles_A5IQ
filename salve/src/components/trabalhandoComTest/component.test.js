import React from 'react'
import renderer from 'react-test-renderer';
import  { ComponenteDeTeste } from './component'

describe('Testando component', () =>{
  it('Mostra componente', () => {
    const component  = (<ComponenteDeTeste a={10} b={20} />)
    const tree = renderer
      .create(component)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})