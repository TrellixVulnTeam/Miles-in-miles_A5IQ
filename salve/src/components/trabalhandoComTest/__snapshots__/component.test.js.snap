// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando component Mostra componente 1`] = `
<div
  style={
    Object {
      "background": "red",
      "fontSize": "20px",
      "height": "200px",
      "margin": "0 auto",
      "marginTop": "100px",
      "padding": "10px",
      "width": "200px",
    }
  }
>
  <div>
    Texto 
    Titulo
    : 
    <br />
     Numero: 
    1000
  </div>
  <a
    href="#!"
    onMouseEnter={[Function]}
  >
    Google | click
  </a>
   
  <br />
  <a
    href="#!"
    onClick={[Function]}
  >
    Bing | click
  </a>
   
  <br />
  <a
    href="http://google.com.br"
  >
    Google
  </a>
</div>
`;
